crates/sovereign-sdk/rollup-interface/src/state_machine/soft_confirmation.rs

```diff
@@ -82,23 +83,23 @@ impl UnsignedSoftConfirmation {
 
 /// Signed version of the `UnsignedSoftConfirmation`
 /// Contains the signature and public key of the sequencer
-#[derive(Debug, PartialEq, Clone, BorshDeserialize, BorshSerialize, Serialize, Deserialize, Eq)]
-pub struct SignedSoftConfirmation {
+#[derive(Debug, PartialEq, BorshDeserialize, BorshSerialize, Serialize, Deserialize, Eq)]
+pub struct SignedSoftConfirmation<'txs> {
     l2_height: u64,
     hash: [u8; 32],
     prev_hash: [u8; 32],
     da_slot_height: u64,
     da_slot_hash: [u8; 32],
     da_slot_txs_commitment: [u8; 32],
     l1_fee_rate: u128,
-    txs: Vec<Vec<u8>>,
+    txs: Cow<'txs, [Vec<u8>]>,
```

Comment:
Because we either own the data (from db/rpc request) or we borrow it (from UnsignedSoftConfirmation). So it's an Rc with strong ref = 1, and no weak refs at all. We don't need Arc, because we don't share it between threads.